packages/components/src/components/sequence/DiagramSpec.ts

```diff
@@ -100,4 +102,30 @@ export default class DiagramSpec {
     if (!parent) throw Error(`No parent found for action spec ${actionSpec.index}`);
     return parent;
   }
+
+  determineVisuallyReachableActors(collapsedActions: boolean[]) {
+    const result = new Set<Actor>();
+
+    for (const actionSpec of this.actions.filter(a => a.nodeType !== 'return')) {
+      // If all of the actors are visible at this point
+      // no need to check remaining actions.
+      if (result.size == this.actions.length)
+        break;
+
+      // If non of the ancestors of this action is collapsed
+      // then this action is visible (not swallowed) and its actors
+      // need to be visible.
+      if (!actionSpec.isCollapsed(collapsedActions)) {
+        const [calle, caller] = actionActors(actionSpec.action);
+
+        if (calle && !result.has(calle))
+          result.add(calle);
```

Comment:
This is a bit nit-picky but it should be `callee` instead of `calle`. Can you change it just so that it's easier to read?